dbms/src/Flash/Pipeline/Schedule/ThreadPool/TaskThreadPool.cpp

```diff
@@ -104,9 +121,16 @@ void TaskThreadPool<Impl>::handleTask(TaskPtr & task)
         if (!Impl::isTargetStatus(status_after_exec) || total_time_spent >= YIELD_MAX_TIME_SPENT_NS)
             break;
     }
-    task_queue->updateStatistics(task, status_before_exec, total_time_spent);
     metrics.addExecuteTime(task, total_time_spent);
     metrics.decExecutingTask();
+    // updateStatistics may throw exception, so need to update metrics first.
```

Comment:
Why does `updateStatistics` throw an error? This will cause tiflash crash

Reply:
take()/updateStatistics()/submit() will handle exception inside.